### ISSUE TITLE
View class androidx.appcompat.widget.AppCompatTextView is an AppCompat widget that can only be used with a Theme.AppCompat theme

### ISSUE THREAD

**Issue by alexgrusu** [aws-amplify/amplify-flutter#253]
**Describe the bug**

I'm facing a problem that was widely discussed in here https://stackoverflow.com/questions/21814825/.

My style.xml file consists the following configuration.

```
<?xml version="1.0" encoding="utf-8"?>
<resources>
<!-- Theme applied to the Android Window while the process is starting -->
<style name="LaunchTheme" parent="Theme.AppCompat.Light">
    <!-- Show a splash screen on the activity. Automatically removed when
         Flutter draws its first frame -->
    <item name="android:windowBackground">@drawable/launch_background</item>
</style>
<!-- Theme applied to the Android Window as soon as the process has started.
     This theme determines the color of the Android Window while your
     Flutter UI initializes, as well as behind your Flutter UI while its
     running.

     This Theme is only used starting with V2 of Flutter's Android embedding. -->
<style name="NormalTheme" parent="Theme.AppCompat.Light">
    <item name="android:windowBackground">@android:color/white</item>
</style>
```
I already tried various themes: Theme.AppCompat.Light, Theme.AppCompat, Theme.AppCompat.Light, Theme.AppCompat.NoActionBar, Theme.AppCompat.Light.NoActionBar, but without any effect.

Finally, I determined what generates the errors.

I'm using amplify in my app and the configuration was implemented as described in here https://docs.amplify.aws/start/getting-started/integrate/q/integration/flutter.

I figured out that if I comment the amplify configuration, the errors are not generated anymore.

**To Reproduce**
There is no specific action that could trigger the errors below, the events are triggered suddenly. Also, when the problem occurs, a menu called "Amplify Developer Menu" appear on the screen.

**Platform**
Android

**Device**
Huawei P9 lite.

**Screenshots**
![Screenshot_20201214-225354](https://user-images.githubusercontent.com/49151614/102134933-dcabe480-3e5f-11eb-8089-67033e7e55de.png)
![Screenshot_20201214-225403](https://user-images.githubusercontent.com/49151614/102134983-f2210e80-3e5f-11eb-948e-dd6d75fd2dc5.png)

**Additional context**
Below are the errors that occur when amplify configuration is allowed. 
```
E/HW-JPEG-DEC(32740): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
E/HW-JPEG-DEC(32740): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
E/HW-JPEG-DEC(32740): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
E/HW-JPEG-DEC(32740): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
E/HW-JPEG-DEC(32740): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
E/ThemeUtils(32740): View class androidx.appcompat.widget.AppCompatTextView is an AppCompat 
widget that can only be used with a Theme.AppCompat theme (or descendant).
W/ResourceType(32740): Too many attribute references, stopped at: 0x01010034
E/ThemeUtils(32740): View class androidx.appcompat.widget.AppCompatImageButton is an 
AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).

```

**Comment by Amplifiyer** [comment]
@alexgrusu can you share your code snippet on how you are calling configure, your pubspec.yaml or better if you have a reproduction repository where we can reproduce this bug?

**Comment by alexgrusu** [comment]
`import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:wakelock/wakelock.dart';
import 'package:amplify_core/amplify_core.dart';
import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
import 'amplifyconfiguration.dart';
import 'screens/login.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  Wakelock.enable(); //< lock sleeping
  SystemChrome.setEnabledSystemUIOverlays([]); // Full screen mode
  runApp(MaterialApp(
    home: MyBuddy(),
  ));
}

class MyBuddy extends StatefulWidget {
  static String route = "mybuddy";

  @override
  State<StatefulWidget> createState() {
    return _MyBuddyState();
  }
}

class _MyBuddyState extends State<MyBuddy> {
  bool _amplifyConfigured = false;

  // Instantiate Amplify
  Amplify amplifyInstance = Amplify();

  @override
  void initState() {
    super.initState();
    _configureFirebase();
    _configureAmplify();
  }

  void _configureFirebase() async {
    await Firebase.initializeApp();
  }

  void _configureAmplify() async {
    if (!mounted) return;

    try {
      AmplifyAuthCognito authPlugin = AmplifyAuthCognito();
      amplifyInstance.addPlugin(authPlugins: [authPlugin]);

      // Once Plugins are added, configure Amplify

      await amplifyInstance.configure(amplifyconfig);

      setState(() {
        _amplifyConfigured = true;
        print('Amplify configured status $_amplifyConfigured');
      });
    } catch (e) {
      print(e);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        decoration: BoxDecoration(
          image: DecorationImage(
            image: AssetImage("assets/background.jpg"),
            fit: BoxFit.cover,
          ),
        ),
        child: LoginScreen(),
      ),
    );
  }
}
`

`name: mybuddy
description: mop

publish_to: 'none' # Remove this line if you wish to publish to pub.dev

https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^1.0.0
  audioplayers: ^0.16.2
  just_audio: ^0.5.5
  firebase_core: ^0.5.2
  cloud_firestore: ^0.14.3
  connectivity: ^2.0.2
  hexagon: ^0.0.4
  wakelock: ^0.2.1+1
  syncfusion_flutter_gauges: ^18.3.52 # Used in progress animation
  amplify_core: '<1.0.0'
  amplify_auth_cognito: '<1.0.0'

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:

  uses-material-design: true

  assets:
    - assets/
`

`import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

final kBoxDecorationStyle = BoxDecoration(
  color: Color(0xFF6CA8F1),
  borderRadius: BorderRadius.circular(10.0),
  boxShadow: [
    BoxShadow(
      color: Colors.black12,
      blurRadius: 6.0,
      offset: Offset(0, 2),
    ),
  ],
);

class LoginData {
  final String name;
  final String password;

  LoginData({
    @required this.name,
    @required this.password,
  });

  @override
  String toString() {
    return '$runtimeType($name, $password)';
  }
}

class LoginScreen extends StatefulWidget {
  static String route = "home";

  @override
  State<StatefulWidget> createState() {
    return _LoginScreenState();
  }
}

class _LoginScreenState extends State<LoginScreen> {
  Widget _buildEmailTF() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: <Widget>[
        Text(
          'Email',
        ),
        SizedBox(height: 10.0),
        Container(
          alignment: Alignment.centerLeft,
          decoration: kBoxDecorationStyle,
          height: 60.0,
          child: TextField(
            keyboardType: TextInputType.emailAddress,
            style: TextStyle(
              color: Colors.white,
            ),
            decoration: InputDecoration(
              border: InputBorder.none,
              contentPadding: EdgeInsets.only(top: 14.0),
              prefixIcon: Icon(
                Icons.email,
                color: Colors.white,
              ),
              hintText: 'Enter your Email',
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildPasswordTF() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: <Widget>[
        Text(
          'Password',
        ),
        SizedBox(height: 10.0),
        Container(
          alignment: Alignment.centerLeft,
          decoration: kBoxDecorationStyle,
          height: 60.0,
          child: TextField(
            obscureText: true,
            style: TextStyle(
              color: Colors.white,
            ),
            decoration: InputDecoration(
              border: InputBorder.none,
              contentPadding: EdgeInsets.only(top: 14.0),
              prefixIcon: Icon(
                Icons.lock,
                color: Colors.white,
              ),
              hintText: 'Enter your Password',
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildForgotPasswordBtn() {
    return Container(
      alignment: Alignment.centerRight,
      child: FlatButton(
        onPressed: () => print('Forgot Password Button Pressed'),
        padding: EdgeInsets.only(right: 0.0),
        child: Text(
          'Forgot Password?',
        ),
      ),
    );
  }

  Widget _buildLoginBtn() {
    return Container(
      padding: EdgeInsets.symmetric(vertical: 25.0),
      width: double.infinity,
      child: RaisedButton(
        elevation: 5.0,
        onPressed: () {
          SystemChrome.restoreSystemUIOverlays();
          print('LOGIN');
        },
        padding: EdgeInsets.all(15.0),
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(30.0),
        ),
        color: Colors.white,
        child: Text(
          'LOGIN',
          style: TextStyle(
            color: Color(0xFF527DAA),
            letterSpacing: 1.5,
            fontSize: 18.0,
            fontWeight: FontWeight.bold,
          ),
        ),
      ),
    );
  }

  Widget _buildSignupBtn() {
    return GestureDetector(
      onTap: () => print('Sign Up Button Pressed'),
      child: RichText(
        text: TextSpan(
          children: [
            TextSpan(
              text: 'Don\'t have an Account? ',
              style: TextStyle(
                color: Colors.white,
                fontSize: 18.0,
                fontWeight: FontWeight.w400,
              ),
            ),
            TextSpan(
              text: 'Sign Up',
              style: TextStyle(
                color: Colors.white,
                fontSize: 18.0,
                fontWeight: FontWeight.bold,
              ),
            ),
          ],
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        SystemChrome.restoreSystemUIOverlays();
        FocusScope.of(context).unfocus();
      },
      child: Container(
        height: double.infinity,
        child: SingleChildScrollView(
          physics: AlwaysScrollableScrollPhysics(),
          padding: EdgeInsets.symmetric(
            horizontal: 40.0,
            vertical: 120.0,
          ),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Text(
                'Sign In',
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 30.0,
                  fontWeight: FontWeight.bold,
                ),
              ),
              SizedBox(height: 30.0),
              _buildEmailTF(),
              SizedBox(
                height: 30.0,
              ),
              _buildPasswordTF(),
              _buildForgotPasswordBtn(),
              _buildLoginBtn(),
              _buildSignupBtn(),
            ],
          ),
        ),
      ),
    );
  }
}
`

**Comment by alexgrusu** [comment]
I think I figured out something. It could be dependent of calling unfocus and restore in GestureDetector.
```

GestureDetector(
      onTap: () {
          SystemChrome.restoreSystemUIOverlays();
          FocusScope.of(context).unfocus();
      },

```
I added a protection based on a keyboard listener.
Looks like it doesn't happen anymore.
```
GestureDetector(
      onTap: () {
        if (keyon) {
          SystemChrome.restoreSystemUIOverlays();
          FocusScope.of(context).unfocus();
        }
      },
```
I will continue the tests and coming back with a feedback.

**Comment by alexgrusu** [comment]
Still occurs, but I can't figure out the cause or how to reproduce it..

**Comment by alexgrusu** [comment]
> Still occurs, but I can't figure out the cause or how to reproduce it..
> 
> For example, I removed the GestureDetector, so that the build method of the LoginScreen returns a Container.
> 
> After I built a new app and starting to debug while taping the text-field and cancelling the keyboard with its 'return' key, the log below results:
> 
> ```
> Flutter run key commands.
> r Hot reload. 🔥🔥🔥
> R Hot restart.
> h Repeat this help message.
> d Detach (terminate "flutter run" but leave application running).
> c Clear the screen
> q Quit (terminate the application on the device).
> An Observatory debugger and profiler on HUAWEI VNS L21 is available at: http://127.0.0.1:43399/8yl_fQruCvg=/
> I/flutter (20534): Amplify configured status true
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/art     (20534): Do partial code cache collection, code=25KB, data=29KB
> I/art     (20534): After code cache collection, code=25KB, data=29KB
> I/art     (20534): Increasing code cache capacity to 128KB
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): false
> I/HwSecImmHelper(20534): mSecurityInputMethodService is null
> I/flutter (20534): true
> E/HW-JPEG-DEC(20534): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
> E/HW-JPEG-DEC(20534): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
> E/HW-JPEG-DEC(20534): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
> E/HW-JPEG-DEC(20534): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
> E/HW-JPEG-DEC(20534): [HME_JPEG_DEC_Delete](3321): HME_JPEG_DEC_Delete: decoder_ctx=null
> E/ThemeUtils(20534): View class androidx.appcompat.widget.AppCompatTextView is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
> W/ResourceType(20534): Too many attribute references, stopped at: 0x01010034
> E/ThemeUtils(20534): View class androidx.appcompat.widget.AppCompatImageButton is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
> I/HwPointEventFilter(20534): do not support AFT because of no config
> ```
> 
> 'True' is printed when the keyboard appears on the screen by activating the textfield, while 'false' is printed when I cancel the keyboard.
> 
> While writing my last comment, the phone was connected and again the error was generated without doing something special.
> 
> ```
> I/flutter (20534): false
> W/ConnectionTracker(20534): Exception thrown while unbinding
> W/ConnectionTracker(20534): java.lang.IllegalArgumentException: Service not registered: lt@d76e680
> W/ConnectionTracker(20534): 	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1407)
> W/ConnectionTracker(20534): 	at android.app.ContextImpl.unbindService(ContextImpl.java:1657)
> W/ConnectionTracker(20534): 	at android.content.ContextWrapper.unbindService(ContextWrapper.java:648)
> W/ConnectionTracker(20534): 	at ci.f(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):1)
> W/ConnectionTracker(20534): 	at ci.d(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):2)
> W/ConnectionTracker(20534): 	at lu.E(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):9)
> W/ConnectionTracker(20534): 	at le.a(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):3)
> W/ConnectionTracker(20534): 	at eg.run(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):3)
> W/ConnectionTracker(20534): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
> W/ConnectionTracker(20534): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
> W/ConnectionTracker(20534): 	at iz.run(:com.google.android.gms.dynamite_measurementdynamite@204713051@20.47.13 (040408-0):5)
> E/ThemeUtils(20534): View class androidx.appcompat.widget.AppCompatTextView is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
> W/ResourceType(20534): Too many attribute references, stopped at: 0x01010034
> E/ThemeUtils(20534): View class androidx.appcompat.widget.AppCompatImageButton is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
> I/HwPointEventFilter(20534): do not support AFT because of no config
> ```
> 
> The "Exception thrown while unbinding" is nothing important as I understood, it will be handled in future.
> 
> The style that I use is :
> 
> ```
> <style name="LaunchTheme" parent="Theme.AppCompat.Light.DarkActionBar">
> <style name="NormalTheme" parent="Theme.AppCompat.Light.DarkActionBar">
> ```

This time I removed the amplify configuration.
I tried the same scenario, taping the text-field and canceling the keyboard, the errors are not shown anymore.
I still think that the errors are related to something that happen when I enable amplify configuration.

**Comment by jamesonwilliams** [comment]
Related: https://github.com/aws-amplify/amplify-flutter/pull/266 which disables the Android Developer Menu from Flutter.

**Comment by Amplifiyer** [comment]
disabling the Android Developer Menu from Flutter is released today in version 0.0.2-dev.2. Let us know if you are still facing any issues. Closing this.
